### DISCUSSION TITLE
Parchment Entrypoint - Crash on Run

### DISCUSSION THREAD

**Discussion by Banjomann** [FabricMC/fabric-api#1789] [Mod Dev Support]
Hi there! New to the community, and I've been trying to spin up a proper mod for a while. Only big issue I've had thus far is this:

It's during bootup after my mod init, during the entry point for parchment. I figured maybe it was one of my recipes or something but they don't look to be the issue (I had the same set of recipes working before changes). I can't seem to get any farther crash dump or trace to determine where it's actually an issue. I can pull apart the entry point, but I figured I'd come here to ask first.

Crash dump:

[parchmentDump.txt](https://github.com/FabricMC/fabric/files/7404153/parchmentDump.txt)

This one's got me scratching my head. Recently installed the Cotton and LibGUI libraries to make stuff easier, seems to be when it started.

Following are my build, properties, and mod.json files. Let me know if I need to upload anything else:

[build.gradle.txt](https://github.com/FabricMC/fabric/files/7404155/build.gradle.txt)
[gradle.properties.txt](https://github.com/FabricMC/fabric/files/7404156/gradle.properties.txt)
[fabric.mod.json.txt](https://github.com/FabricMC/fabric/files/7404157/fabric.mod.json.txt)

**Comment by Banjomann** [comment]
Didn't mention, I'm on the latest (#IC-212.5457.46) build of IntelliJ. Seemed to be working fine on eclipse.

**Comment by warjort** [comment]
````
Caused by: java.lang.ClassNotFoundException: jdk.nashorn.api.scripting.NashornScriptEngineFactory
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:636)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:182)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:175)
````
Your problem is that cottonmc/parchment uses Nashorn (a javascript engine).
It assumes this is available by default, but it was removed in Java 15
https://openjdk.java.net/jeps/372

You need to find a way to put Nashorn back in the classpath.
